packages/themes/ecl/src/ecl-ec/mixins/button.scss

```diff
@@ -1,102 +1,107 @@
+@import '../../mixins/rem';
+
 @mixin kol-button {
-	a,
-	button {
+	.kol-button {
+		$root: &;
+
 		appearance: none;
-		background-color: none;
 		outline: none;
 		text-decoration: none;
-	}
+		&:focus {
+			outline-offset: rem(-4);
+			outline: var(--text-focus-outline-color) solid rem(2);
+		}
 
-	a:focus .kol-span-wc,
-	button:focus .kol-span-wc {
-		outline-offset: rem(-4);
-		outline-width: rem(2);
-		outline-style: solid;
-	}
+		--text-focus-outline-color: var(--color-black);
 
-	button.ghost:focus .kol-span-wc {
-		outline-offset: rem(-2);
-	}
+		&--primary {
+			--text-background-color: var(--color-blue);
+			--text-border-color: var(--color-blue);
+			--text-color: var(--color-white);
 
-	.primary a .kol-span-wc,
-	.primary button .kol-span-wc {
-		background-color: var(--color-blue);
-		border-color: var(--color-blue);
-		color: var(--color-white);
-	}
+			&:focus {
+				--text-focus-outline-color: var(--color-white);
+			}
 
-	.primary a:focus .kol-span-wc,
-	.primary button:focus .kol-span-wc {
-		outline-color: var(--color-white);
-	}
+			&:hover {
+				--text-background-color: var(--color-blue-130);
+				--text-border-color: var(--color-blue-130);
+			}
+		}
 
-	.primary a:hover .kol-span-wc,
-	.primary button:hover .kol-span-wc {
-		background-color: var(--color-blue-130);
-		border-color: var(--color-blue-130);
-	}
+		&--secondary {
+			--text-background-color: var(--color-white);
+			--text-border-color: var(--color-blue);
+			--text-color: var(--color-blue);
 
-	.secondary a .kol-span-wc,
-	.secondary button .kol-span-wc {
-		background-color: var(--color-white);
-		border-color: var(--color-blue);
-		color: var(--color-blue);
-	}
+			&:focus {
+				--text-focus-outline-color: var(--color-blue);
+				.kol-button__text {
```

Comment:
Könnten wir diese Styles noch nach unten, in den eigentlichen __text-Block schieben?